pkg/snapstore/snapstore_test.go

```diff
@@ -278,10 +280,281 @@ var _ = Describe("Save, List, Fetch, Delete from mock snapstore", func() {
+								var credentialFullPaths []string
+								for i := range credentialFileNames {
+									// file at fileSkipIndex is skipped from being created
+									if i == fileSkippedIndex {
+										continue
+									}
+									credentialFullPaths = append(credentialFullPaths, filepath.Join(credentialDirectory, credentialFileNames[i]))
+								}
+								createCredentialFiles(credentialFullPaths)
+							})
+							It("Should return error", func() {
+								newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(providerToSnapstoreProviderMap[provider])
+								Expect(err).Should(HaveOccurred())
+								Expect(newSecretModifiedTime.IsZero()).Should(Equal(true))
+							})
+						})
+					}
+					Describe("Credential files exist", func() {
+						var lastCreationTimeForFile time.Time
+						var credentialFileFullPaths []string
+						BeforeEach(func() {
+							for i := range credentialFileNames {
+								credentialFileFullPaths = append(credentialFileFullPaths, filepath.Join(credentialDirectory, credentialFileNames[i]))
+							}
+							lastCreationTimeForFile = createCredentialFiles(credentialFileFullPaths)
+						})
+						Context("Files not modified", func() {
+							It("Should return creation timestamp", func() {
+								newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(providerToSnapstoreProviderMap[provider])
+								Expect(err).ShouldNot(HaveOccurred())
+								Expect(newSecretModifiedTime.Equal(lastCreationTimeForFile)).Should(Equal(true))
+							})
+						})
+						// test with each particular credential file modified
+						for fileModifiedIndex := 0; fileModifiedIndex < len(credentialFileNames); fileModifiedIndex++ {
+							fileModifiedIndex := fileModifiedIndex
+							Context("Files have been modified", func() {
+								BeforeEach(func() {
+									fmt.Println("Testing for the provider:", providerToSnapstoreProviderMap[provider])
+									bytesRead, err := os.ReadFile(credentialFileFullPaths[fileModifiedIndex])
+									if err != nil {
+										fmt.Println("Error while reading the unmodified credential file:", err)
+									}
+									fmt.Printf("The original contents of the file are: \"%s\"\n", string(bytesRead))
+									err = modifyCredentialFile(fileModifiedIndex, credentialFileFullPaths)
+									Expect(err).ShouldNot(HaveOccurred())
+								})
+								It("Should return modification timestamp", func() {
+									newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(providerToSnapstoreProviderMap[provider])
+									Expect(err).ShouldNot(HaveOccurred())
+									fmt.Println("Creation time for the credential file:\t\t", lastCreationTimeForFile)
+									fmt.Println("Modification time for the credential file:\t", newSecretModifiedTime)
+									// printing contents of the file
+									bytesRead, err := os.ReadFile(credentialFileFullPaths[fileModifiedIndex])
+									if err != nil {
+										fmt.Println("Error while reading the modified credential file:", err)
```

Comment:
same as above

Reply:
Addressed in [0da0475](https://github.com/gardener/etcd-backup-restore/pull/670/commits/0da04759cb89ded3f2cb7b9f4f203e734d3310d4)